framework/include/base/SubProblem.h

```diff
@@ -368,6 +400,12 @@ class SubProblem : public Problem
   /* This needs to remain <unsigned int> for threading purposes */
   std::vector<unsigned int> _has_active_elemental_moose_variables;
 
+  /// Set of material property ids that determine whether materials get reinited
+  std::vector<std::set<unsigned int>> _active_material_property_ids;
+
+  /// True if _active_material_property_ids is not empty
+  std::vector<unsigned int> _has_active_material_property_ids;
```

Comment:
There are certain members of the team that are terrified and completely against vector<bool> because it has bitten them (us) before in threaded cases. You won't find it much in the framework because of that. I'm not part of that group but it's just not worth fighting for the most part.

He's following existing patterns here. We can clean them all up later.